encoding/box_query_encoding.go

```diff
@@ -0,0 +1,33 @@
+package encoding
+
+import (
+	"encoding/base64"
+	"github.com/algorand/go-algorand-sdk/client/v2/common/models"
+	"github.com/algorand/go-algorand-sdk/types"
+)
+
+const encodingBase64Prefix = "b64:"
+
+// EncodeBytesForBoxQuery provides a convenience method to string encode box names for use with Box search APIs (e.g. GetApplicationBoxByName).
+func EncodeBytesForBoxQuery(xs []byte) string {
+	return encodingBase64Prefix + base64.StdEncoding.EncodeToString(xs)
+}
+
+// EncodeBoxForBoxQuery provides a convenience method to string encode box names for use with Box search APIs (e.g. GetApplicationBoxByName).
+func EncodeBoxForBoxQuery(b models.Box) (string, error) {
+	decoded, err := base64.StdEncoding.DecodeString(string(b.Name))
```

Comment:
If `models.Box.Name` is meant to be used as a b64 string then why is the type bytes? This PR doesn't need to be concerned with this though, I'm just curious.

Reply:
@algoidurovic It's a good question - let me know if the response addresses it?

* The data type is bytes to match the OAS2 specification for type = _string_ + format = _bytes_  (https://swagger.io/specification/v2/ - see _Data Types_).
* It's definitely confusing.  Your question led me to closely (re-)review go-algorand.  And I briefly paired with @algochoi (thanks!) to clarify the types.